src/Components/Game.tsx

```diff
@@ -0,0 +1,182 @@
+import React, { useState } from "react";
+import Board from "./Board";
+import { GameProps, isNotNull } from "../Types/TypeProps";
+import GameInfo from './GameInfo';
+/**
+ * Handle Game logic
+ * @param props 
+ * @returns 
+ */
+const Game = (props:GameProps) => {
+  /**
+   * Game's state declarations
+   */
+  const [history, setHistory] = useState(props.historyVal);
+  const [stepNumber, setStepNumber] = useState(props.stepNumberVal);
+  const [xIsNext, setXisNext] = useState(props.xIsNextVal);
+  const [counter, setCounter] = useState(history.length);
+
+  /**
+   * Handle each board's cell action when clicked
+   * @param {*} i
+   */
+  const handleClick = (i: number) => {
+    const gameHistory = history.slice(0, stepNumber + 1);
+    const current = gameHistory[gameHistory.length - 1];
+    const squares = current.squares.slice();
+    
+
+    // Prevent further cell clicks after winning
+    if (calculateWinner(squares) || squares[i]) return;
+
+    // Determine next player
+    squares[i] = xIsNext ? "X" : "O";
+
+    // Update states
+    setHistory(gameHistory.concat([{ squares: squares }]));
+    setCounter(history.length);
+    setStepNumber(gameHistory.length);
+    setXisNext((prevXisNext) => !prevXisNext);
+  };
+
+  /**
+   * Reset game
+   */
+  const reset = () => {
+    setHistory(props.historyVal);
+    setCounter(1);
+    setStepNumber(props.stepNumberVal);
+    setXisNext(props.xIsNextVal);
+  };
+
+  /**
+   * Go to previous Move
+   */
+  const prevMove = () => {
+    if (stepNumber === 0) return;
+    setStepNumber((prevStep) => prevStep - 1);
+    setXisNext((next) => !next);
+  };
+
+  /**
+   * Go to next move
+   */
+  const nextMove = () => {
+    if (stepNumber >= counter || counter === 1) return;
+    setStepNumber((prevStep) => prevStep + 1);
+    setXisNext((next) => !next);
+  };
+
+  /**
+   * Update time travel steps with step number
+   * @param {*} step
+   */
+  const jumpTo = (step: number) => {
+    setStepNumber(step);
+    setXisNext(step % 2 === 0);
+  };
+
+    /**
+   * Calculate game winner
+   * @param {*} squares
+   */
+  const calculateWinner = (squares: Array<string>) => {
```

Comment:
I would define this outside of the React lifecycle, you have 2 variables with the same name within your scope here (squares). You don't anything from the closure anyway.